internal/parse/parser.go

```diff
@@ -113,14 +121,28 @@ func (p *Parser) add(part queryPart) {
 // a pointer to a ParsedExpr.
 func (p *Parser) Parse(input string) (*ParsedExpr, error) {
 	p.init(input)
+
 	for {
 		p.partStart = p.pos
-		if p.pos == len(p.input) {
+
+		if ip, ok, err := p.parseInputExpression(); err != nil {
+			return nil, fmt.Errorf("parser error: %s", err)
+		} else if ok {
+			p.add(ip)
+
+		} else if sp, ok, err := p.parseStringLiteral(); err != nil {
+			return nil, fmt.Errorf("parser error: %s", err)
+		} else if ok {
+			p.add(sp)
+
+		} else if p.pos == len(p.input) {
 			break
+		} else {
+			p.pos++
```

Comment:
This else is only triggered when theres nothing else starting at that position that can be parsed. The parser then moves on to the next position and tries there. When something is successfully parsed, all the input that was skipped over before getting to it is then added as a BypassPart in `p.add`. Even though its ignoring content, it will be eventually parsed into a BypassPart.

Reply:
Just for the record, we discussed the problem of this approach in a call. It should do a more controlled skip towards the next start point.